pubsub/rabbitmq/rabbitmq.go

```diff
@@ -212,7 +223,32 @@ func (r *rabbitMQ) prepareSubscription(channel rabbitMQChannelBroker, req pubsub
 	}
 
 	r.logger.Debugf("%s declaring queue '%s'", logMessagePrefix, queueName)
-	q, err := channel.QueueDeclare(queueName, true, r.metadata.deleteWhenUnused, false, false, nil)
+	var args amqp.Table
+	if r.metadata.enableDeadLetter {
+		// declare dead letter exchange
+		dlxName := fmt.Sprintf("%s-%s", defaultDeadLetterExchangeNamePrefix, queueName)
```

Comment:
Why not let the DLX and DLQ be free form? Is this convention mandated by RabbitMQ?

Reply:
This is not convention mandated by RabbitMQ. Prefixes（DLX and DLQ） can be customized by metadata. Is this OK? In this case, the queueName is `consumerID-Topic`, the dlx name will be `dlxPrefixFromMetadata-consumerID-Topic`, the dlq name will be `dlqPrefixFromMetadata-consumerID-Topic`.

Reply:
@Taction sorry for the delayed response. What do you think about having the names be formats (e.g. `dlq-%s` -> usage: `dlxName := fmt.Sprintf(defaultDeadLetterExchangeFormat, queueName)`).

This would like you specify a constant or plug the queue name in anywhere you want.  Thoughts?

Reply:
Ping @Taction, we'd like to get this merged.

Reply:
@pkedy fixed.

@yaron2 this is the last review comment needs to be fixed.